common/src/model_registration/register.py

```diff
@@ -129,7 +131,21 @@ def main(args):
             max_version = (max(models_list, key=lambda x: x.version)).version
             model_version = str(int(max_version) + 1)
     except Exception:
-        print(f"Error in listing versions for model {model_name}. Trying to register model with version '1'.")
+        print(
+            f"Error in listing versions for model {model_name}. Trying to register model with version '1'."
+        )
+
+    # Updating tags and properties with value provided in metadata file
+    if args.model_metadata:
+        with open(args.model_metadata, "r") as stream:
+            metadata = yaml.safe_load(stream)
+            tags = metadata.get("tags", tags)
+            properties = metadata.get("properties", properties)
+            model_description = metadata.get("description", model_description)
+
+    # Updating properties from model_info file
+    for key in PROPERTIES:
+        properties[key] = model_info["metadata"]["download_details"][key]
```

Comment:
please take care of nullable scenario here too

Reply:
Done.
As of now I have assumed by checking model_download component that metadata and download_details key will always be there, for keys inside model_info["metadata"]["download_details"] , I have applied the check.

Reply:
Thank you. Resolving